test/jdk/java/foreign/TestLinker.java

```diff
@@ -180,7 +180,12 @@ public void interwovenPadding() {
         var struct = MemoryLayout.structLayout(JAVA_BYTE, padding1, padding2, JAVA_INT);
 
         var fd = FunctionDescriptor.of(struct, struct, struct);
-        assertThrows(IllegalArgumentException.class, () -> linker.downcallHandle(fd));
+        try {
+          linker.downcallHandle(fd);
+        } catch (IllegalArgumentException e) {
+            assertEquals(e.getMessage(),
+                    "The padding layout x2 was preceded by another padding layout x1 in [b1x1x2i4]");
+        }
```

Comment:
Unfortunately not in testng. In JUnit 5 it does.

Reply:
Testng should have this as well, since 6.9.5: https://www.javadoc.io/doc/org.testng/testng/latest/org/testng/Assert.html#expectThrows(java.lang.Class,org.testng.Assert.ThrowingRunnable) jtreg current bundles testng 7.3.0